source/common/http/utility.cc

```diff
@@ -649,6 +649,40 @@ bool Utility::isWebSocketUpgradeRequest(const RequestHeaderMap& headers) {
                                  Http::Headers::get().UpgradeValues.WebSocket));
 }
 
+void Utility::removeUpgrade(RequestOrResponseHeaderMap& headers,
+                            const std::vector<Matchers::StringMatcherPtr>& matchers) {
+  if (headers.Upgrade()) {
+    auto tokens = Envoy::StringUtil::splitToken(headers.getUpgradeValue(), ",", false, true);
```

Comment:
prefer explicit type here. maybe `absl::InlineVector<absl::string_view, 4>` to avoid unnecessary head allocation.